### PR TITLE
Refactor palindrom checking algorithm

### PR DESCRIPTION
**Pull request by bgdnrvsky**
## Description

- Use simpler approach to check if the string is a palindrome
- Add unit test for empty palindrome

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

### CODE REVIEW

File: src/string/burrows_wheeler_transform.rs [hunk 1]
```diff
@@ -42,76 +42,55 @@ pub fn inv_burrows_wheeler_transform<T: AsRef<str>>(input: (T, usize)) -> String
...
+            burrows_wheeler_transform($string)
+        };
+    }
+
+    macro_rules! decipher {
+        ($string:expr) => {
+            inv_burrows_wheeler_transform($string)
+        };
+    }
+
```
Comment (vil02) [inline src/string/burrows_wheeler_transform.rs:56]: What is the advantage of using these macros?

File: src/string/burrows_wheeler_transform.rs
Comment (vil02) [inline src/string/burrows_wheeler_transform.rs:1]: The title of this PR is _Refactor palindrom checking algorithm_. In order to move faster, I would suggest to revert the changes of this file and create another PR with them.

File: src/string/palindrome.rs [hunk 2]
```diff
@@ -1,11 +1,5 @@
 pub fn is_palindrome(s: &str) -> bool {
-    let mut chars = s.chars();
-    while let (Some(c1), Some(c2)) = (chars.next(), chars.next_back()) {
-        if c1 != c2 {
-            return false;
-        }
-    }
-    true
+    s.chars().eq(s.chars().rev())
```
Comment (vil02) [inline src/string/palindrome.rs:2]: This is a nice one liner but quite inefficient (both in terms of space and time). Would you like to rewrite it avowing the copy and avoiding the redundant checks in case of a palindrome?

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/string/burrows_wheeler_transform.rs
Comment (bgdnrvsky) [reply src/string/burrows_wheeler_transform.rs:1]: Yeah, sorry, accidentally pushed it to the `master` branch

File: src/string/palindrome.rs (hunk 2 above)
Comment (bgdnrvsky) [reply src/string/palindrome.rs:2]: Agree with you, at this point it is better to close the PR
